velox/exec/Unnest.cpp

```diff
@@ -146,41 +143,74 @@ RowVectorPtr Unnest::getOutput() {
   return output;
 }
 
+Unnest::RowRange Unnest::extractRowRange(vector_size_t size) const {
+  vector_size_t numInput = 0;
+  vector_size_t numElements = 0;
+  std::optional<vector_size_t> lastRowEnd;
+  for (auto row = nextInputRow_; row < size; ++row) {
+    const bool isFirstRow = (row == nextInputRow_);
+    const vector_size_t remainingSize =
+        isFirstRow ? rawMaxSizes_[row] - firstRowStart_ : rawMaxSizes_[row];
+    ++numInput;
+    if (numElements + remainingSize > maxOutputSize_) {
+      // Single row's output is divided into multiple batches.
+      // Computes the row range to process the first and last row
+      // partially instead of 0 to rawMaxSizes_[row].
```

Comment:
Comment suggestion:
> // A single row's output needs to be split into multiple batches.
// Determines the range to process the first and last rows partially,
// rather than processing from 0 to 'rawMaxSizes_[row]'.